### DISCUSSION TITLE
about testing rigid body collision resolution on CPU in fluid scenarios

### DISCUSSION THREAD

**Discussion by 123uiui** [NVIDIA-Omniverse/PhysX#330] [Q&A]
I am currently working on a project where I would like to test the speed of rigid body collision resolution in fluid scenarios, comparing CPU and GPU performance. Could you kindly guide me on how to set up the rigid body collision resolution to run on the CPU (rather than the GPU) within a fluid simulation?
think you

**Comment by vreutskyy** [comment]
Hi @123uiui 
As far as I know fluid simulation is a GPU-only feature in PhysX. Or do I misunderstand what you're trying to do?

**Reply by 123uiui** [reply]
I'm sorry for not explaining it clearly.
Performance testing

fluid+GPU Rigid Bodies
fluid+CPU Rigid Bodies
And
The simulation of fluids requires the use of a particle system, and the particle system must utilize the GPU.I can test the first scenario, but how should I test the second scenario?how should I set sceneDesc.flags？

**Reply by vreutskyy** [reply]
There's no separate option to enable GPU Rigid Bodies. When you set up the GPU simulation, it means your rigid bodies will be simulated on GPU. And as particles are GPU-only feature, they require GPU simulation. So, there's no way to have both CPU Rigid Bodies and fluid (particle system) on the same scene.

**Reply by 123uiui** [reply]
thank you very much
